de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/ScriptHandler.java

```diff
@@ -377,36 +381,43 @@ private void saveGeneratedResources(FskPortObject fskPortObject, File workingDir
     try {
       File newResourcesDirectory = FileUtil.createTempDir("generatedResources");
 
-      try {
-        if (!command.equals("c()") && !command.equals("print([])")) {
+
+      if (!command.equals("c()") && !command.equals("print([])")) {
+        try {
           String[] filenames = runScript(command, exec, true);
 
           // Copy every resource from the working directory
           for (String filename : filenames) {
-            // file parameters can come from another generatedResourceDirectory
-            // from another model; in that case we use that path since it is
-            // not present in the current workingDirectory
-            Path sourceDir = workingDirectory.toPath();
-            Path source = Paths.get(filename.trim());
-            // if a file variable has a path already attached, use that instead
-            File sourceFile = sourceDir.resolve(source).toFile();
-            File targetFile = new File(newResourcesDirectory, source.getFileName().toString());
-            FileUtil.copy(sourceFile, targetFile, exec);
+            try {
+              // file parameters can come from another generatedResourceDirectory
+              // from another model; in that case we use that path since it is
+              // not present in the current workingDirectory
+              Path sourceDir = workingDirectory.toPath();
+              Path source = Paths.get(filename.trim());
+              // if a file variable has a path already attached, use that instead
+              File sourceFile = sourceDir.resolve(source).toFile();
+              File targetFile = new File(newResourcesDirectory, source.getFileName().toString());
+              FileUtil.copy(sourceFile, targetFile, exec);
+            } catch (CanceledExecutionException | IOException e) {
+              throw new ResourceFileNotFoundException(filename);
+            }
           }
+        } catch (REXPMismatchException | IOException e) {

```

Comment:
runScript is also throwing RException, CanceledExecutionException and InterruptedException.

Reply:
@miguelalba you are right, but why doesn't the compiler complain here?